geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteFunctionOp.java

```diff
@@ -496,11 +501,11 @@ private static List constructAndGetFunctionTasks(final PoolImpl pool, final Stri
       if (isReexecute == 1) {
         resultCollector.clearResults();
       }
+
     }
 
     private void addBytes(byte isReexecute, byte fnStateOrHasResult) {
-      if (ConnectionImpl
-          .getClientFunctionTimeout() == ConnectionImpl.DEFAULT_CLIENT_FUNCTION_TIMEOUT) {
+      if (getTimeoutMs() == DEFAULT_CLIENT_FUNCTION_TIMEOUT) {
```

Comment:
I think the comparison to the DEFAULT constant here is a bit confusing.
It works, currently, because the DEFAULT is 0. But if someone changed the default then this code would be broken.
We only want to encode the timeout if it is not 0 so I think this "if" expression should compare to 0.

Reply:
This same comparison happens in other "addBytes" implementations

Reply:
What about just always encoding it? Is it worth saving the bytes?

Reply:
I think it would be okay to always send it, even when it is 0. You would need to check the server side and make sure it does the correct thing when 0 is sent.

Reply:
Server side it does this:

```java
    int functionTimeout = DEFAULT_CLIENT_FUNCTION_TIMEOUT;
...
      if (bytes.length >= 5
          && serverConnection.getClientVersion().ordinal() >= Version.GFE_8009.ordinal()) {
        functionTimeout = Part.decodeInt(bytes, 1);
      }
```
So if we always send it we are good.

Reply:
The more I think about it the more I want to leave it as is. The `DEFAULT_CLIENT_FUNCTION_TIMEOUT` is defined on both sides, server and client. The server initializes the timeout to `DEFAULT_CLIENT_FUNCTION_TIMEOUT` then reads the timeout if encoded. While it is currently defined as `0` it is possible we can change it and as long as server and client both agree on the value of `DEFAULT_CLIENT_FUNCTION_TIMEOUT` it's ok.

The bug does exist that if the client and server don't agree on `DEFAULT_CLIENT_FUNCTION_TIMEOUT` then they they can have different expected timeouts. This isn't a new issue. The way this is lazily encoded should probably be addressed in a rewrite of the function messages. The current messages look a little Frankenstein internally. There are several byte sized boolean flags and this random collection of bytes that may contain this timeout.

Reply:
I agree with you. Thanks for looking into this.